src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -118,8 +118,17 @@ public static function createConnection($dsn, array $options = array())
         $params += $options + self::$defaultConnectionOptions;
         if (null === $params['class'] && !\extension_loaded('redis') && !class_exists(\Predis\Client::class)) {
             throw new CacheException(sprintf('Cannot find the "redis" extension, and "predis/predis" is not installed: %s', $dsn));
+        } elseif (isset($params['predis_options']) && !class_exists(\Predis\Client::class)) {
```

Comment:
we prefer replacing such "elseif" by an "if" (can be done when the previous "if" break the code flow, with throw or return typically. (same below)